transmart-rest-api/grails-app/controllers/org/transmartproject/rest/DimensionController.groovy

```diff
@@ -70,12 +61,87 @@ class DimensionController extends AbstractQueryController {
         return multiDimensionalDataResource.getDimension(dimensionName)
     }
 
-    private ContainerResponseWrapper wrapElements(Dimension dim, Iterable elements) {
-        def des = new DimensionElementSerializer(dim, elements.iterator())
-        new ContainerResponseWrapper(
-                container: des,
-                key: 'elements',
-                componentType: Object)
+    @CompileStatic
+    private List<DimensionProperties> getDimensions(User user) {
+        return studiesResource.getStudiesWithMinimalPatientDataAccessLevel(user, PatientDataAccessLevel.MEASUREMENTS).stream()
+                .flatMap({ MDStudy study ->
+                    study.dimensions.stream()
+                        .map({ Dimension dimension -> DimensionProperties.forDimension(dimension) }) })
+                .distinct()
+                .collect(Collectors.toList())
+    }
+
+    /**
+     * Dimensions endpoint:
+     * <code>/v2/dimensions</code>
+     *
+     * @return a list of all dimensions that the user has access to.
+     */
+    def list() {
+        def args = getGetOrPostParams()
+        checkForUnsupportedParams(args, ['dimensionName', 'constraint'])
+
+        def dimensions = getDimensions(authContext.user)
+
+        response.status = HttpStatus.OK.value()
+        response.contentType = 'application/json'
+        response.characterEncoding = 'utf-8'
+        new ObjectMapper().writeValue(response.outputStream, [dimensions: dimensions])
+    }
+
+    /**
+     * Dimension endpoint:
+     * <code>/v2/dimensions/${dimensionName}</code>
+     *
+     * @return the dimension properties if the user has access to it.
```

Comment:
I'd expect from this call to return single dimension object instead.
Should this call have `/v2/dimensions/${dimensionName}/properties` path instead?

Reply:
That is exactly what is gives, a single dimension object instead. But not a dimension as in all elements of one dimension of a hypercube (something I don't like about the current Dimension interface: it combines dimension as an object with retrieval of dimension elements), but just its properties.